cmd/rehearse/main.go

```diff
@@ -0,0 +1,222 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"strings"
+
+	// TODO: Solve this properly
+	// "github.com/davecgh/go-spew/spew"
+	"github.com/getlantern/deepcopy"
+	// "github.com/ghodss/yaml"
+	"github.com/sirupsen/logrus"
+
+	pjapi "k8s.io/test-infra/prow/apis/prowjobs/v1"
+	pjclientset "k8s.io/test-infra/prow/client/clientset/versioned"
+	"k8s.io/test-infra/prow/config"
+	prowflagutil "k8s.io/test-infra/prow/flagutil"
+	"k8s.io/test-infra/prow/github"
+	"k8s.io/test-infra/prow/pjutil"
+
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+func getJobsToExecute(ca *config.Agent) config.JobConfig {
+	var youHaveOneJob config.Presubmit
+	for _, job := range ca.Config().Presubmits["openshift/ci-operator"] {
+		if job.Name == "pull-ci-openshift-ci-operator-master-build" {
+			youHaveOneJob = job
+			break
+		}
+	}
+
+	jobs := config.JobConfig{
+		Presubmits: map[string][]config.Presubmit{"openshift/ci-operator": []config.Presubmit{youHaveOneJob}},
+	}
+	return jobs
+}
+
+func makeRehearsalPresubmit(source *config.Presubmit, repo string, prNumber int) (*config.Presubmit, error) {
+	var rehearsal config.Presubmit
+	deepcopy.Copy(&rehearsal, source)
+
+	rehearsal.Name = fmt.Sprintf("rehearse-%d-%s", prNumber, source.Name)
+	rehearsal.Context = fmt.Sprintf("ci/rehearse/%s/%s", repo, strings.TrimPrefix(source.Context, "ci/prow/"))
+
+	if len(source.Spec.Containers) != 1 {
+		return nil, fmt.Errorf("Cannot rehearse jobs with more than 1 container in Spec")
+	}
+	container := source.Spec.Containers[0]
+
+	if len(container.Command) != 1 || container.Command[0] != "ci-operator" {
+		return nil, fmt.Errorf("Cannot rehearse jobs that have Command different from simple 'ci-operator'")
+	}
+
+	for _, arg := range container.Args {
+		if strings.HasPrefix(arg, "--git-ref") || strings.HasPrefix(arg, "-git-ref") {
+			return nil, fmt.Errorf("Cannot rehearse jobs that call ci-operator with '--git-ref' arg")
+		}
+	}
+
+	if len(source.Branches) != 1 {
+		return nil, fmt.Errorf("Cannot rehearse jobs that run over multiple branches")
+	}
+	branch := strings.TrimPrefix(strings.TrimSuffix(source.Branches[0], "$"), "^")
```

Comment:
Oh boy

Reply:
Cute, aint' it? The items in that list are regexes, so in some cases we might not be able to extract the "proper" name from it without asking GH for a list of all branches on the target repo and checking whether the job would run on that branch via the regex. You think it's worth the effort when we actually force our org to have a single brach with `^master$`-like regexes?

Reply:
No. Agreed we can just handle the single branch cases because those are the interesting ones